pkg/controller/utils/utils.go

```diff
@@ -274,62 +271,28 @@ func GetAmazonCloudIntegration(ctx context.Context, client client.Client) (*oper
 	return instance, nil
 }
 
-// getAuthentication
-func GetAuthentication(ctx context.Context, client client.Client) (interface{}, error) {
-	var authenticationConfig interface{}
+// GetAuthentication finds the authentication CR in your cluster.
+func GetAuthentication(ctx context.Context, cli client.Client) (*operatorv1.Authentication, error) {
+	authentication := &operatorv1.Authentication{}
+	err := cli.Get(ctx, DefaultTSEEInstanceKey, authentication)
+	if err != nil {
+		return nil, err
+	}
+	if authentication.Spec.OIDC != nil && authentication.Spec.Openshift != nil {
```

Comment:
Ah, I think this is the validation I was looking for. 

Please see this similar comment about the pattern we've established for defaulting / validation in the operator. I think it mostly applies here as well: https://github.com/tigera/operator/pull/941#discussion_r509548336

Basically:

- Query the object
- Fill in defaults for missing fields.
- Validate the object
- Write it back.
- Render components
- Update status

Any other controllers that depend on Dex / the Authentication resource should monitor for the .Status.Ready field, which indicates all of this logic has been completed.